File: Top 100 Codes/03-Codes for Recursion/13-Given a positive integer N, return the Nth row of pascal’s triangle/GivenapositiveintegerNreturntheNthrowofpascalstriangle.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> getNthRow(int n) {
    vector<int> row(n + 1, 1);  // Initialize all elements as 1

    for (int i = 1; i < n; ++i) {
        row[i] = (row[i - 1] * (n - i + 1)) / i;  // Compute using binomial coefficient
    }

    return row;
}

int main() {
    int n;
    cout << "Enter the row number (0-indexed): ";
    cin >> n;

    vector<int> nthRow = getNthRow(n);

    cout << "The " << n << "th row of Pascal's Triangle is: ";
    for (int num : nthRow) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}

File: Top 100 Codes/03-Codes for Recursion/13-Given a positive integer N, return the Nth row of pascal’s triangle/GivenapositiveintegerNreturntheNthrowofpascalstriangle.py
def get_nth_row(n):
    row = [1]  # Initialize the first element as 1
    for i in range(1, n + 1):
        row.append(row[i - 1] * (n - i + 1) // i)  # Calculate using binomial coefficient
    return row

# Example usage
n = int(input("Enter the row number (0-indexed): "))
nth_row = get_nth_row(n)
print(f"The {n}th row of Pascal's Triangle is: {nth_row}")
